Allowing relative paths in URDF

FracturedShader opened the pull request:
# 🦟 Bug fix
## Summary
When parsing a URDF file, the `_source` parameter is set to "urdf file" which breaks relative path resolution later on. This commit changes the call to `sdf::readDoc` to use `filename` as the `_source` parameter instead.

## Additional Information
I came across this issue while testing out a physics plugin I am developing for PhysX 4.1 integration. Searching all repos pulled by colcon reveals no other uses of the phrase "urdf file" as a special case. Testing has revealed no new issues.

azeey commented:
Thanks for the contribution. Since this is a small change, would you mind setting the target branch to `sdf9` so the bug fix is goies into earlier versions as well?

FracturedShader commented:
> Thanks for the contribution. Since this is a small change, would you mind setting the target branch to `sdf9` so the bug fix is goies into earlier versions as well?

No problem. Rebased onto `sdf9` and modified the commit to fit.
